lib/index.js

```diff
@@ -403,36 +460,66 @@ exports.getFunctionUrl = async (ctx, boundaryId, functionId) => {
 
 exports.getStorageClient = (ctx) => {
     const expiry = 60 * 16; // 15+1 min to align with Lambda lifecycle plus some buffer
-    const accessToken = Jwt.sign({}, Buffer.from(ctx.configuration.fusebit_storage_key, 'base64').toString('utf8'), {
-        algorithm: 'RS256',
-        expiresIn: expiry,
-        audience: ctx.configuration.fusebit_storage_audience,
-        issuer: ctx.configuration.fusebit_storage_issuer_id,
-        subject: ctx.configuration.fusebit_storage_subject,
-        keyid: ctx.configuration.fusebit_storage_key_id,
-        header: { jwtId: Date.now().toString() },
-    });
+    let accessToken = ctx.fusebit && ctx.fusebit.functionAccessToken;
+    if (ctx.configuration.fusebit_storage_key) {
+        // Legacy mode - use pre-creaeted issuer to mint the access token
+        accessToken = Jwt.sign({}, Buffer.from(ctx.configuration.fusebit_storage_key, 'base64').toString('utf8'), {
+            algorithm: 'RS256',
+            expiresIn: expiry,
+            audience: ctx.configuration.fusebit_storage_audience,
+            issuer: ctx.configuration.fusebit_storage_issuer_id,
+            subject: ctx.configuration.fusebit_storage_subject,
+            keyid: ctx.configuration.fusebit_storage_key_id,
+            header: { jwtId: Date.now().toString() },
+        });
+    }
 
     const url = generateStorageUrl(ctx.configuration);
 
-    return {
+    const isHierarchicalStorageSupported = accessToken === ctx.fusebit.functionAccessToken;
+
+    const ensureHierarchicalStorageSupported = () => {
+        if (!isHierarchicalStorageSupported) {
+            throw new Error('Hierarchical storage is not supported in this function.');
+        }
+    }
+
+    const getUrl = (storageSubId) => `${url}${storageSubId ? ('/' + storageSubId) : ''}`;
+
+    const storageClient = {
         etag: null,
         expiration: Date.now() + expiry * 1000,
-        get: async function () {
-            const response = await Superagent.get(url)
+        get: async function (storageSubId) {
+            storageSubId && ensureHierarchicalStorageSupported();
+            const response = await Superagent.get(getUrl(storageSubId))
                 .set('Authorization', `Bearer ${accessToken}`)
                 .ok((res) => res.status < 300 || res.status === 404);
             this.etag = response.body.etag;
             return response.status === 404 ? undefined : response.body.data;
         },
-        put: async function (data, force) {
-            let request = Superagent.put(url).set('Authorization', `Bearer ${accessToken}`);
+        put: async function (data, force, storageSubId) {
```

Comment:
If we do go with Benn's suggestion to rev the major version and break, I'd leave `force` last in this method

Reply:
Agreed. I don't think reving major makes sense given the limited lifetime of this piece.

Reply:
Done.